client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -69,8 +71,21 @@ public String toString() {
                 '}';
     }
 
-    static Request ping() {
-        return new Request("HEAD", "/", Collections.emptyMap(), null);
+    // For developers please add your new methods in the alphabetical order
+
+    static Request delete(DeleteRequest deleteRequest) {
+        String endpoint = endpoint(deleteRequest.index(), deleteRequest.type(), deleteRequest.id());
+
+        Params parameters = Params.builder();
+        parameters.withRouting(deleteRequest.routing());
+        parameters.withParent(deleteRequest.parent());
+        parameters.withTimeout(deleteRequest.timeout());
+        parameters.withVersion(deleteRequest.version());
+        parameters.withVersionType(deleteRequest.versionType());
+        parameters.withRefreshPolicy(deleteRequest.getRefreshPolicy());
+        parameters.withWaitForActiveShards(deleteRequest.waitForActiveShards());
+
+        return new Request(HttpDelete.METHOD_NAME, endpoint, parameters.getParams(), null);
```

Comment:
Do you want to test the Params builder actually?

Something like:

```java
    public void testWithParams() {
        Request.Params params = Request.Params.builder();
        params.withFetchSourceContext(FetchSourceContext.FETCH_SOURCE);
        params.withParent("parent");
        params.withPipeline("pipeline");
        params.withPreference("preference");
        params.withRealtime(true);
        if (randomBoolean()) {
            params.withRefresh(true);
        } else {
            params.withRefreshPolicy(WriteRequest.RefreshPolicy.WAIT_UNTIL);
        }
        params.withRouting("routing");
        params.withStoredFields(new String[]{"field1","field2"});
        params.withTimeout(TimeValue.timeValueMillis(100));
        params.withVersion(2);
        params.withVersionType(VersionType.EXTERNAL);
        params.withWaitForActiveShards(ActiveShardCount.ALL);
        Map<String, String> requestParams = params.getParams();
        assertEquals(10, requestParams.size());
    }
```

I can add it in the PR or later.